test/compflags/ferguson/ldflags-order.prediff

```diff
@@ -6,19 +6,28 @@ LOG=$2
 #    executable>, <log>, <compiler executable>)
 
 MYLOG=""
-
 LAST=""
+FOUND_MARKER=0
+
+while read line || [[ -n $line ]];
+do
+  # skip until we find "compiling generated source" line
+  if [[ $line == *"compiling generated source"* ]]; then
+    FOUND_MARKER=1
+  fi
+  if [ $FOUND_MARKER -ne 1 ]; then continue; fi
 
-for word in `cat $LOG`
-  do
-    if [[ $word == -Ltest-* ]]
-      then
-        if [[ $word != $LAST ]]
-          then
-            MYLOG+=" $word"
-            LAST=$word
-          fi
-      fi
-done
+  for word in $line
+    do
```

Comment:
Does this technically provide an arbitrary code execution path for people running ldflags-order test? That would just mean we don't want to allow CI to run by default with this test, if that were ever a thing.

Reply:
I believe I've fixed this now, but am not experienced enough in Bash to be sure it's safe to run on untrusted input. However, I think that's alright as we'd have a lot of other precautions to worry about if we were going to allow CI to run our test suite; there are surely other `.prediff`s with similar issues, and in any case a user could PR changes to the `.prediff`s themselves to execute arbitrary code if we weren't careful.